sycl/doc/extensions/experimental/sycl_ext_oneapi_work_group_scratch_memory.asciidoc

```diff
@@ -0,0 +1,188 @@
+buffer without passing it as an argument to the kernel.
+Device local memory is memory that is shared by all work-items in a work-group.
+The behavior is similar to the usage of unbounded array with the CUDA `+__shared__+` keyword.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_WORK_GROUP_DYNAMIC` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+
+=== `get_work_group_scratch_memory` function
+
+The `get_work_group_scratch_memory` function provides access
+to a dynamically allocated buffer in the device local memory.
+
+[source,c++]
+----
+void* get_work_group_scratch_memory()
+----
+_Constraints_: `T` must be trivially constructible and trivially destructible.
+
+_Returns_: A pointer to a dynamically allocated buffer
+           in the device local memory.
+
+The size of the allocation is unknown at compile-time,
+and must be communicated to the SYCL implementation via the
+`work_group_scratch_size` property. Every call to
+`get_work_group_scratch_memory` returns the same allocation
+in device local memory.
+
+=== Kernel properties
+
+The `work_group_scratch_size` property must be passed to a kernel to determine
+the run-time size of the device local memory allocation associated with
+all `get_work_group_scratch_memory` calls.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+struct work_group_scratch_size {
+  constexpr work_group_scratch_size(size_t bytes) : value(bytes) {}
+  size_t value;
+}; // work_group_scratch_size
+
+using work_group_scratch_size_key = work_group_scratch_size;
+
+template <> struct is_property_key<work_group_scratch_size_key> : std::true_type {};
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+|===
+|Property|Description
+
+|`work_group_scratch_size`
+|The `work_group_scratch_size` property describes the amount of dynamic
+device local memory required by the kernel in bytes.
+
+|===
+
+=== Total allocation check
+
+If the total amount of device local memory requested (i.e., the sum of
+all memory requested by `local_accessor`, `group_local_memory`,
+`group_local_memory_for_overwrite`, `work_group_static` and `work_group_scratch_size`) exceeds a device's
+local memory capacity (as reported by `local_mem_size`) then the implementation
+must throw a synchronous `exception` with the `errc::memory_allocation` error
+code from the kernel invocation command (e.g. `parallel_for`).
+
+==== Usage examples
```

Comment:
```suggestion
== Example
```

The "Example" section should be a peer of "Specification" because the examples are not a normative part of the specification.

Reply:
indeed, fixed both extensions